backend/tracim_backend/lib/webdav/dav_provider.py

```diff
@@ -1,33 +1,140 @@
+            current_user=current_user, session=session, config=app_config
+        )
+        self.content_api = ContentApi(current_user=current_user, session=session, config=app_config)
+
+        self.workspaces = deque()
+        self.contents = deque()
+        path_parts = self._path_splitter(self.path)
+        if len(path_parts) >= 1:
+            root_workspace_filemanager_filename = webdav_convert_file_name_to_bdd(path_parts[0])
+            try:
+                root_workspace = self.workspace_api.get_one_by_filemanager_filename(
+                    root_workspace_filemanager_filename
+                )
+            except WorkspaceNotFound:
+                root_workspace = None
+            self.workspaces.append(root_workspace)
+            # INFO - G.M - 2020-10-07 - Get all workspace tree + content_tree
+            if root_workspace:
+                search_sub_workspace = True
+                for path_part in path_parts[1:]:
+                    section = webdav_convert_file_name_to_bdd(path_part)
+                    if search_sub_workspace:
+                        try:
+                            workspace = self.workspace_api.get_one_by_filemanager_filename(
+                                section, parent=self.workspaces[-1]
+                            )
+                            self.workspaces.append(workspace)
+                            continue
+                        except WorkspaceNotFound:
+                            search_sub_workspace = False
+                            pass
+                    try:
+                        content = self.content_api.get_one_by_filename(
+                            filename=section,
+                            workspace=self.workspaces[-1],
+                            parent=self.contents[-1] if self.contents else None,
+                        )
+                    except ContentNotFound:
+                        content = None
+                    self.contents.append(content)
+
+    def _path_splitter(self, path: str) -> typing.List[str]:
+        path_parts = path.split("/")
+        # removing leading or trailing empty path part
+        if not path_parts[0]:
+            path_parts = path_parts[1:]
+        if path_parts and not path_parts[-1]:
+            path_parts = path_parts[:-1]
+        return path_parts
+
+    @property
+    def current_workspace(self) -> typing.Optional[Workspace]:
+        try:
+            return self.workspaces[-1]
+        except IndexError:
+            return None
+
+    @property
+    def current_content(self) -> typing.Optional[Content]:
+        try:
+            return self.contents[-1]
+        except IndexError:
+            return None
+
+    @property
+    def current_parent_content(self) -> typing.Optional[Content]:
+        try:
+            return self.contents[-2]
+        except IndexError:
+            return None
+
+    @property
+    def current_path_level(self) -> typing.Optional[ResourceLevelType]:
```

Comment:
`level` made me think it would be the depth of the current path (ie. an integer), should the naming be `current_path_type`/`current_path_resource_type`?

Reply:
done.